crates/client/src/rr/dnssec/key_format.rs

```diff
@@ -387,8 +387,11 @@ mod tests {
         let encoded = key_format.generate_and_encode(algorithm, en_pass);
 
         if encode {
-            assert!(encoded.is_ok(), format!("{}", encoded.unwrap_err()));
-            let decoded = key_format.decode_key(&encoded.unwrap(), de_pass, algorithm);
+            let encoded = match encoded {
+                Ok(x) => x,
+                Err(x) => panic!("{}", x)
```

Comment:
This would be simpler to write as `let encoded = encoded.expect("error in encoding")`. Did clippy not recommend that for some reason?